homework-14/main.js

```diff
@@ -0,0 +1,121 @@
+// 1
+const outputYear = document.getElementById("input1");
+const getCentury = document.getElementById("btn");
+const message = document.getElementById("result");
+
+getCentury.addEventListener("click", () => {
+  message.innerHTML = `Entered ${outputYear.value} year is ${yearToCentury(+outputYear.value)} century`;
+});
+
+function yearToCentury(year) {
+  let century = Math.floor(year / 100) + 1;
+
+  switch (year % 10) {
+    case 1:
+      century = century + "st";
+      break;
+    case 2:
+      century = century + "nd";
+      break;
+    case 3:
+      century = century + "rd";
+      break;
+    default:
+      century = century + "th";
+  }
+  return century;
+}
+
+// 2
+const enteredStr = document.getElementById("input2");
+const getString = document.getElementById("btn2");
+const outputNumber = document.getElementById("output-number");
+
+getString.addEventListener("click", () => {
+  outputNumber.innerHTML = `The result of ${enteredStr.value} is ${vowelOne(
+    enteredStr.value
+  )}`;
+});
+
+function vowelOne(str) {
+  const vowels = ["a", "e", "i", "o", "u"];
+  let result = "";
+  for (let letter of str) {
+    if (vowels.includes(letter)) {
+      result += "1";
+    } else {
+      result += "0";
+    }
+  }
+  return result;
+}
+
+// 3
+const enteredPhrase = document.getElementById("input3");
+const getPhrase = document.getElementById("btn3");
+const reversedSubString = document.getElementById("reversed");
+
+getPhrase.addEventListener("click", () => {
+  reversedSubString.innerHTML = `It is a new str ${spinWords(
+    enteredPhrase.value
+  )}`;
+});
+
+function spinWords(str) {
+  let newString = "";
+  let words = str.split(" ");
+
+  words.forEach((element) => {
+    if (element.length >= 5) {
+      newString +=
+        (newString ? " " : "") + element.split("").reverse().join("");
+    } else {
+      newString += (newString ? " " : "") + element;
+    }
+  });
+  return newString;
+}
+
+// 4
+const inputNumber = document.getElementById("input4");
+const getNumber = document.getElementById("btn4");
+const outputResult = document.getElementById("high-and-low");
+
+getNumber.addEventListener("click", () => {
+  outputResult.innerHTML = `From array of numbers the highest ${numbers[0]} 

```

Comment:
Here and below ` numbers ` is an unresolved variable, and that's why you have an error in the console

<img width="655" alt="image" src="https://user-images.githubusercontent.com/84010501/200839741-1ccc4a08-2a4c-4ef1-9490-3120bca8510b.png">